task/control-plane-agent/src/mgs_sidecar.rs

```diff
@@ -393,6 +405,65 @@ impl MgsHandler {
     }
 }
 
+fn verify_signature(
+    data: &EcdsaSha2Nistp256Challenge,
+    key: &[u8; 65],
+    signer_nonce: &[u8; 8],
+    signature: &[u8; 64],
+) -> Result<(), GwMonorailError> {
+    if !TRUSTED_KEYS.iter().any(|t| t == key) {
+        ringbuf_entry!(Trace::WrongKey);
+        return Err(GwMonorailError::UnlockAuthFailed);
+    }
+
+    let sig = p256::ecdsa::Signature::from_bytes(signature.as_slice().into())
+        .map_err(|_| GwMonorailError::UnlockAuthFailed)?;
+
+    let v = p256::ecdsa::VerifyingKey::from_encoded_point(
+        &p256::EncodedPoint::from_bytes(key)
+            .map_err(|_| GwMonorailError::UnlockAuthFailed)?,
+    )
+    .map_err(|_| GwMonorailError::UnlockAuthFailed)?;
+
+    // Build an SSH signature blob to be verified
```

Comment:
Is there not a rust crate to do this or is there one but it's not `no-std`

Reply:
[`ssh_key::SshSig::signed_data`](https://docs.rs/ssh-key/latest/ssh_key/struct.SshSig.html#method.signed_data) is what we want, but it returns a `Vec<u8>`.  This is because the namespace (`monorail-unlock` here) is of unknown size.